eth/downloader/downloader.go

```diff
@@ -344,6 +351,11 @@ func (d *Downloader) synchronise(id string, hash common.Hash, entropy *big.Int,
 	if d.headEntropy.Cmp(entropy) >= 0 {
 		return nil
 	}
+
+	// Only start the downloader after we reset from a forked state
+	if d.core.BadHashExistsInChain() {
+		return nil
```

Comment:
Do we want to print a warning here (if not return an error). Say that something went wrong and bad hashes persist in the chain. Would explain to the user why they're not downloading blocks.

Reply:
Yes I can add a warning print